src/main/java/com/mongodb/spark/sql/connector/read/partitioner/SamplePartitioner.java

```diff
@@ -121,21 +122,24 @@ public SamplePartitioner() {}
     }
 
     int numberOfSamples = (int) Math.ceil((samplesPerPartition * count) / numDocumentsPerPartition);
-
+    Bson projection =
+        partitionField.contains(ID_FIELD)
```

Comment:
It seems that we must use `equals` instead.

Reply:
Done - was a hangover from the field list. good spot.